processor/src/chiplets/aux_trace/mod.rs

```diff
@@ -382,126 +384,129 @@ fn build_bitwise_request<E: FieldElement<BaseField = Felt>>(
         + alphas[4].mul_base(z)
 }
 
-/// Builds `MLOAD*` and `MSTORE*` requests made to the memory chiplet.
-fn build_mem_request<E: FieldElement<BaseField = Felt>>(
+/// Builds `MLOAD` and `MSTORE` requests made to the memory chiplet.
+fn build_mem_request_element<E: FieldElement<BaseField = Felt>>(
     main_trace: &MainTrace,
     op_label: u8,
-    word: bool,
     alphas: &[E],
-    i: usize,
+    row: usize,
 ) -> E {
-    let ctx = main_trace.ctx(i);
-    let clk = main_trace.clk(i);
-
-    let (v0, v1, v2, v3) = if word {
-        (
-            main_trace.stack_element(0, i + 1),
-            main_trace.stack_element(1, i + 1),
-            main_trace.stack_element(2, i + 1),
-            main_trace.stack_element(3, i + 1),
-        )
-    } else {
-        (
-            main_trace.helper_0(i),
-            main_trace.helper_1(i),
-            main_trace.helper_2(i),
-            main_trace.stack_element(0, i + 1),
-        )
-    };
+    let word = [
+        main_trace.stack_element(0, row + 1),
+        main_trace.helper_register(2, row),
+        main_trace.helper_register(1, row),
+        main_trace.helper_register(0, row),
```

Comment:
What is stored in these helper registers, and why is it included as part of the request computation? Can we add a comment that explains it?

This would help understand the `value` argument to `compute_memory_request()`, which is currently unclear to me (since it's not always "the word that was read or written").

And based on this new understanding, I might rename `build_mem_request_{element, word}()`.

Reply:
I infer from `build_memory_chiplet_response()` that the helper registers must hold the remaining 3 elements that were read/written. 

I would 

- make that clear in a comment here
- In `compute_memory_request()`'s docstring, I would add a sentence to say that `value` is the memory value that was read/written
- rename `build_mem_request_element()` -> `build_loadstore_request()`
- rename `build_mem_request_word()` -> `build_loadstorew_request()`

I suggest renaming `build_mem_request_{element, word}()` yet again because I had something else in mind when I suggested it. I thought we would have these 2 basic functions that everything else would depend on (which is now `compute_memory_request()`). Now these 2 are associated exclusively with the `MLOAD/STORE{W}` instructions, and so I would rename them so as to capture that.